### ISSUE TITLE
getText() returns text breaking up words with spaces

### ISSUE THREAD

**Issue by kotsis** [smalot/pdfparser#201]
I use as example the file http://www.pdf995.com/samples/pdf.pdf
getText() returns this:

"The p df9 95  s u it e  o f p ro ducts  -  P df9 95,  P dfE dit 9 95 , a n d S ig na tu re 99 5 -  is  a  c o m ple te  s o lu tio n fo r y o ur d ocu m ent p ublis h in g n ee ds. It 
p ro vid es e ase  o f u se , fle xib ilit y  in  fo rm at,  a nd  in d ustr y -s ta nd ard  s e cu rit y - a nd a ll  a t n o  c o st to  y o u. 
P df9 95	 m ake s it  e a sy a nd  a ffo rd able  to  c re ate  p ro fe ssio nal- q u alit y  d o cu m en ts  in  th e p o pula r P D F file  fo rm at. I ts  e asy-to -u se  in te rfa ce  
h elp s y o u to  c re a te  P D F file s b y s im ply  s e le ctin g t h e " p rin t"  c o m mand f r o m  a ny a pp lic a tio n,  c re atin g d ocu m ents  w hic h  c a n  b e v ie w ed 
o n a ny c o m pute r w it h  a  P D F v ie w er. P df9 95 s u pp orts  n etw ork  file  s a vin g, f a st u se r s w it c h in g  o n X P , C it r ix /T erm in al  S erv e r, c u sto m  
p age s iz e s a nd la rg e fo rm at  p rin tin g . P df9 95 is  a  p rin te r d riv e r th at w ork s w it h  a ny P osts crip t to  P D F c o nve rte r. T he p df9 95  p rin te r 
d riv e r a nd a  fr e e C onve rte r a re  a va ila b le  fo r e a sy d ow nlo ad . 	
P dfE dit 9 95	 o ffe rs  a  w ealt h  o f a ddit io na l  fu n ctio na lit y , s u ch  a s: c o m bin in g d o cu m en ts  in to  a  s in g le  P D F;  a uto m atic  lin k in se rtio n; 
h ie ra rc h ic a l  b ookm ark  in se rtio n ; P D F c o nve rs io n to  H TM L o r D O C ( te xt o nly ); in te gra tio n w it h  W ord  to olb ar w it h  a u to m atic  ta ble  o f 
c o nte nts  a nd lin k g enera tio n ; a u to atta ch  to  e m ail;  s ta tio ne ry  a n d s ta m pin g.  	
S ig natu re 99 5	 o ffe rs  s ta te -o f- th e-a rt s e cu rit y  a n d e n cry p tio n t o  p ro te ct y o u r d ocu m ents  a n d a dd d ig it a l  s ig n atu re s. 	
    	.... etc"

Is there something wrong with the pdf? 

I tried the same pdf with pdftotext and I get the text without the spaces.

Best regards,
Kostas

**Comment by davidfindlay** [comment]
I'm having this same problem and it seems to be one of kerning. It's not good at sensing spaces and non-spaces.

**Comment by JoeCodeTech** [comment]
I have the same problem. I tried the demo on the pdfparser.org, and input a one page pdf file which was a basic business invoice, and it showed many of the words with spaces between each character, as the example above also shows. Some text fields were OK, and some were not. This happened with text that was in mixed case, and also in capitals-only case.

**Comment by reza-akbari** [comment]
same issue, any workaround of update?

**Comment by fycben** [comment]
I have had this problem for a while and just found a solution someone else had, and worked for me and a lot of other people.

https://github.com/smalot/pdfparser/issues/72#issuecomment-159574323

**Comment by Reqrefusion** [comment]
There is the same problem with the pdf files created by Wikipedia.
Sample:
[Sarayönü_Camii.pdf](https://github.com/smalot/pdfparser/files/5365907/Sarayonu_Camii.pdf)

**Comment by Reqrefusion** [comment]
Less but similar problem in the following file.
Examples:
BAŞLANGI Ç
M illetinin
duygularını n
h er
sevg i
File:
https://www.mevzuat.gov.tr/MevzuatMetin/1.5.2709.pdf
It is not the subject of the title, but the quotation marks are incorrect in the related file (&#34; )

**Comment by Reqrefusion** [comment]
A similar problem is happening here.
https://www.yok.gov.tr/Documents/Mevzuat/kirgizistan-turkiye-manas-universitesi-tuzugu.pdf

**Comment by owsygavin** [comment]
I'm having the same issue.

I am creating PDFs with Gotenberg:

```
<html>
<head>
</head>
<body>
  <h1>There are no forms configured.</h1>
</body>
</html>
```

They render just fine in-browser, but when I read them back in via PdfParser and call `getText()`, `There are no forms configured.` becomes `There are no forms configur ed.` (inserting a space).

Using the node package https://www.npmjs.com/package/pdf-parse , I get the behaviour I would expect - it renders `There are no forms configured.`

So I think it probably is something to do with kerning

**Comment by rubenvanerk** [comment]
Using #505, this seems to fix the issue:

```php
$config = new \Smalot\PdfParser\Config();
$config->setHorizontalOffset('');

$parser = new \Smalot\PdfParser\Parser([], $config);
$pdf    = $parser->parseFile('document.pdf');
 
$text = $pdf->getText('path/to/pdf.pdf');
```

Outputs:
```text
The pdf995 suite of products - Pdf995, PdfEdit995, and Signature995 - is a complete solution for your document publishing needs. It 
provides ease of use, flexibility in format, and industry-standard security- and all at no cost to you. 
Pdf995	 makes it easy and affordable to create professional-quality documents in the popular PDF file format. Its easy-to-use interface 
helps you to create PDF files by simply selecting the "print" command from any application, creating documents which can be viewed 
on any computer with a PDF viewer. Pdf995 supports network file saving, fast user switching on XP, Citrix/Terminal Server, custom 
page sizes and large format printing. Pdf995 is a printer driver that works with any Postscript to PDF converter. The pdf995 printer 
driver and a free Converter are available for easy download. 	
PdfEdit995	 offers a wealth of additional functionality, such as: combining documents into a single PDF; automatic link insertion; 
hierarchical bookmark insertion; PDF conversion to HTML or DOC (text only); integration with Word toolbar with automatic table of 
contents and link generation; autoattach to email; stationery and stamping.  	
Signature995	 offers state-of-the-art security and encryption to protect your documents and add digital signatures. 

...
```

**Comment by k00ni** [comment]
I merged #505 into master branch. It will be part of the next release.
